### PR TITLE
Add missing extern for `Array.from` (copy of #1335)

### PR DESCRIPTION
**Pull request by Dominator008**

### CODE REVIEW

**Comment by Dominator008** [comment]
I authored these commits.

**Comment by Dominator008** [comment]
@MatrixFrog Rebased here.

File: externs/es6.js [hunk 1]
```diff
@@ -1166,6 +1166,16 @@ Promise.prototype.then = function(opt_onFulfilled, opt_onRejected) {};
...
 
 /**
+ * @param {string|!IArrayLike<T>|!Iterable<T>} arrayLike
+ * @param {?function(this:S, T, number, string|!IArrayLike<T>|!Iterable<T>): R} opt_mapFn
+ * @param {S=} opt_this
+ * @return {!Array<R>}
+ * @template T,S,R
+ * @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from
+ */
+Array.from = function(arrayLike, opt_mapFn, opt_this) {};
```
Comment (MatrixFrog) [inline externs/es6.js]: We already have this at line 1192. Just add the missing params there.

Comment (Dominator008) [inline externs/es6.js]: Done.

File: externs/es6.js [hunk 2]
```diff
@@ -1173,13 +1174,16 @@ Promise.prototype.catch = function(onRejected) {};
...
 Array.of = function(var_args) {};
 
+
 /**
  * @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from
- * @param {!Array<T>|!IArrayLike<T>|!Iterable<T>} arrayLike
- * @return {!Array<T>}
- * @template T
+ * @param {string|!Array<T>|!IArrayLike<T>|!Iterable<T>} arrayLike
+ * @param {?function(this:S, T, number, string|!Array<T>|!IArrayLike<T>|!Iterable<T>): R} opt_mapFn
```
Comment (Dominator008) [inline externs/es6.js]: Is this line too long?

Comment (MatrixFrog) [inline externs/es6.js]: Ugh sadly yes.

Comment (Dominator008) [inline externs/es6.js]: Done.

**Comment by MatrixFrog** [comment]
lgtm, just waiting for travis...

**Comment by MatrixFrog** [comment]
Can you update the author/committer metadata to be accurate so that googlebot is happy?

**Comment by Dominator008** [comment]
Done.

**Pull request merged**